mmdet3d/models/necks/dla_neck.py

```diff
@@ -0,0 +1,193 @@
+                 norm_cfg=None):
+        super(DLAUp, self).__init__()
+        self.start_level = start_level
+        if in_channels is None:
+            in_channels = channels
+        self.channels = channels
+        channels = list(channels)
+        scales = np.array(scales, dtype=int)
+        for i in range(len(channels) - 1):
+            j = -i - 2
+            setattr(
+                self, 'ida_{}'.format(i),
+                IDAUp(channels[j], in_channels[j:], scales[j:] // scales[j],
+                      norm_cfg))
+            scales[j + 1:] = scales[j]
+            in_channels[j + 1:] = [channels[j] for _ in channels[j + 1:]]
+
+    def forward(self, layers):
+        """Forward function.
+
+        Args:
+            layers (tuple[torch.Tensor]): features from multi-scale layers.
+        """
+        outs = [layers[-1]]
+        for i in range(len(layers) - self.start_level - 1):
+            ida = getattr(self, 'ida_{}'.format(i))
+            ida(layers, len(layers) - i - 2, len(layers))
+            outs.insert(0, layers[-1])
+        return outs
+
+
+@NECKS.register_module()
+class DLA_Neck(nn.Module):
+    """DLA Neck.
+
+    Args:
+        in_channels (list[int]): List of input channels of multi-scale
+                                feature map.
+        start_level (int): the scale level where upsampling starts. Default: 2.
+        end_level (int): the scale level where upsampling ends. Default: 5.
+        norm_cfg (dict): Config dict for normalization layer. Default: None.
+    """
+
+    def __init__(self,
+                 in_channels=[16, 32, 64, 128, 256, 512],
+                 start_level=2,
+                 end_level=5,
+                 norm_cfg=None):
+        super(DLA_Neck, self).__init__()
+        self.start_level = start_level
+        self.end_level = end_level
+        scales = [2**i for i in range(len(in_channels[self.start_level:]))]
+        self.dla_up = DLAUp(
+            start_level=self.start_level,
+            channels=in_channels[self.start_level:],
+            scales=scales,
+            norm_cfg=norm_cfg)
+        self.ida_up = IDAUp(
+            in_channels[self.start_level],
+            in_channels[self.start_level:self.end_level],
+            [2**i for i in range(self.end_level - self.start_level)], norm_cfg)
+
+    def forward(self, x):
+        x = list(x)
+        x = self.dla_up(x)
+        outs = []
+        for i in range(self.end_level - self.start_level):
+            outs.append(x[i].clone())
+        self.ida_up(outs, 0, len(outs))
+        return outs[-1]
+
+    def init_weights(self):
+        for m in self.modules():
+            if isinstance(m, nn.ConvTranspose2d):
+                # In order to be consistent with the source code,
+                # reset the ConvTranspose2d initialization parameters
+                m.reset_parameters()
+                # Simulated bilinear upsampling kernel
+                w = m.weight.data
```

Comment:
We'd better abstract this init operation to a separate function (as you've done last time `fill_up_conv_weight`). Because in the future we may find this init function useless or we may want to adopt other init function. This can be more flexible.